src/test/groovy/com/example/demo/CarTests.groovy

```diff
@@ -0,0 +1,262 @@
+            def e3 = new CarEntity()
+                e3.setLocation(loc)
+                e3.setManufacturer("BMW")
+                carEntityRepository.save(e3)
+
+        when: "controller is called for alphabetical list"
+            def result = mockMvc.perform(get("/api/car?sort=manufacturer")).andReturn().response.contentAsString
+            def json = new JsonSlurper().parseText(result)
+        then: "Check if objects are in the correct order"
+            json[0].manufacturer == e2.manufacturer
+            json[1].manufacturer == e3.manufacturer
+            json[2].manufacturer == e1.manufacturer
+
+           (json[0].manufacturer.compareTo(json[1].manufacturer) <= 0)
+           (json[2].manufacturer.compareTo(json[1].manufacturer) >= 0)
+
+        cleanup:
+            deleteTestCar(e1.id)
+            deleteTestCar(e2.id)
+            deleteTestCar(e3.id)
+            deleteTestLocation(loc.id)
+    }
+
+    def "Update a location"()
+    {
+        given: "an entity"
+            def loc = locationEntityRepository.save(new LocationEntity())
+            def e1 = new CarEntity()
+                e1.setLocation(loc)
+                e1.setManufacturer("Mercedes")
+                carEntityRepository.save(e1)
+        when: "controller is called with an id, and new manufacturer"
+            def result = mockMvc.perform(put("/api/car/{id}",e1.id).contentType(MediaType.APPLICATION_JSON_VALUE).content("{\"manufacturer\":\"Audi\",\"location\":{\"id\":${loc.id}}}")).andReturn().response.contentAsString
+            def json = new JsonSlurper().parseText(result)
+        then: "verify the manufacturer has changed to Audi"
+            json.manufacturer == "Audi"
+        cleanup:
+            deleteTestCar(e1.id)
+            deleteTestLocation(loc.id)
+
+    }
+
+
+    def "Delete a location"()
+    {
+        given: "two entities"
+            def loc = locationEntityRepository.save(new LocationEntity())
+            def e1 = new CarEntity()
+                e1.setLocation(loc)
+                carEntityRepository.save(e1)
+            def e2 = new CarEntity()
+                e2.setLocation(loc)
+                carEntityRepository.save(e2)
+        when: "controller is called with an id, delete entity"
+            mockMvc.perform(delete("/api/car/{id}",e1.id).contentType(MediaType.APPLICATION_JSON))
+        then: "check if the entity still exists"
+            !carEntityRepository.findById(e1.id)
+            carEntityRepository.findById(e2.id)
+        cleanup:
+            deleteTestCar(e2.id)
+            deleteTestLocation(loc.id)
+
+
+
+    }
+
+    def "deleteTestLocation"(def id)
+    {
+        return mockMvc.perform(delete("/api/location/{id}",id).contentType(MediaType.APPLICATION_JSON)).andReturn().response.getContentAsString()
+    }
+    def "deleteTestCar"(def id)
+    {
+        return mockMvc.perform(delete("/api/car/{id}",id).contentType(MediaType.APPLICATION_JSON)).andReturn().response.getContentAsString()
+    }
```

Comment:
👍 this! We have a bad habit of not cleaning up some of our test repos and it can impact further tests. You'll find some of our test repos have util classes that provide helper methods like these.